src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,582 @@
+                    {
+                        if (editor.OriginalRoot.Contains(nodeToReplace))
+                        {
+                            var replacementNodeWithTrivia = kvp.Value
+                                .WithLeadingTrivia(nodeToReplace.GetLeadingTrivia())
+                                .WithTrailingTrivia(nodeToReplace.GetTrailingTrivia());
+                            editor.ReplaceNode(nodeToReplace, replacementNodeWithTrivia);
+                        }
+                    }
+                }
+
+                return editor.GetChangedRoot();
+            }
+
+            /// <summary>
+            /// Generate a dictionary which key is the symbol in Callee (include local variable and parameter), value
+            /// is the replacement syntax node for all its occurence.
+            /// </summary>
+            private static ImmutableDictionary<ISymbol, SyntaxNode> ComputeReplacementTable(
+                AbstractInlineMethodRefactoringProvider<TInvocationNode, TExpression, TArgumentSyntax, TMethodDeclarationSyntax, TIdentifierName>
+                    inlineMethodRefactoringProvider,
+                IMethodSymbol calleeMethodSymbol,
+                ImmutableArray<(IParameterSymbol parameterSymbol, TExpression literalExpression)> parametersWithLiteralArgument,
+                ImmutableArray<IParameterSymbol> parametersWithDefaultValue,
+                SyntaxGenerator syntaxGenerator,
+                ImmutableDictionary<ISymbol, string> renameTable)
+            {
+                // 1. Type arguments (generics)
+                // Example:
+                // Before:
+                // void Caller()
+                // {
+                //     Callee<int>();
+                // }
+                // void Callee<T>() => Print(typeof<T>);
+                // After:
+                // void Caller()
+                // {
+                //     Callee<int>();
+                // }
+                // void Callee<T>() => Print(typeof<int>);
+                var typeParametersReplacementQuery = calleeMethodSymbol.TypeParameters
+                    .Zip(calleeMethodSymbol.TypeArguments,
+                        (parameter, argument) => (parameter: (ISymbol)parameter,
+                            syntaxNode: inlineMethodRefactoringProvider.GenerateTypeSyntax(argument)));
+                // 2. Literal replacement
+                // Example:
+                // Before:
+                // void Caller()
+                // {
+                //     Callee(20)
+                // }
+                // void Callee(int i, int j = 10)
+                // {
+                //     Bar(i, j);
+                // }
+                // After:
+                // void Caller()
+                // {
+                //     Callee(20)
+                // }
+                // void Callee(int i, int j = 10)
+                // {
+                //     Bar(20, 10);
+                // }
```

Comment:
why is teh callee changing?

Reply:
:) I might not express that clearly but that's one snapshot, basically I want to change the content in that snapshot, then cut the changed method body back and paste it in the document.  I can update the comments.